src/main/java/com/google/step/servlets/HomePageServlet.java

```diff
@@ -246,7 +261,42 @@ private String userNotLoggedIn(String homePageSection) {
       if (limit > 0) {
         deals = deals.stream().limit(limit).collect(Collectors.toList());
       }
+    } else if (sort.equals(DISTANCE_SORT)) {
```

Comment:
I don't know what's the bug, but when i go to 
/api/home?section=restaurants&sort=distance&latitude=1.29351&longitude=103.2
it works, but when i go to
/api/home?section=restaurants&sort=distance&latitude=1.29351&longitude=103
it returns an empty array

Reply:
I went to check some reverse geocoding apparently the lat 1.29351 and long 103 is invalid i guess they need decimals?